### PR TITLE
Core initialization change, removal of Core.Platform.Shared

### PR DESCRIPTION
**Pull request by wojta**
## Motivation

Simplifying initialization of the Core. Other projects than Xamarin.Forms apps were not previously supported.

## Description

We need to move code from MobileCoreInitializer to the new platform specific initializer classes.

## Progress

- [x] Removal of Core.Platform.Shared
- [x] initializers for Android, iOS and Xamarin.Forms

### Xamarin.Forms platform-independent app project (like Example app)

`mobile-services.json` is there in `Resources` directory of the platform-independent app project, configured build action as *Embedded resource*

in Android `Activity` where Xamarin.Forms are initialized you add:
```C#
var app = new App();
MobileCoreAndroid.Init(app.GetType().Assembly,ApplicationContext);
LoadApplication(app);
```
in iOS `AppDelegate` where Xamarin.Forms are initialized you add:
```C#
 var xamApp = new App();
MobileCoreIOS.Init(app.GetType().Assembly);
LoadApplication(xamApp);
```
### Android app written in C# Xamarin

`mobile-services.json` is there in `Assets` directory of the Android Xamarin project, configured build action as *AndroidAsset*

in Android override `Application` class and do:
```C#
MobileCoreAndroid.Init(ApplicationContext);
```

### iOS app written in C# Xamarin

`mobile-services.json` is there in `Resources` directory of the Android Xamarin project, configured build action as *BundleResource*

in iOS just use in your `AppDelegate` or entry-point
```C#
MobileCoreIOS.Init();
```

### CODE REVIEW

File: Core/Core.Platform.Android/AndroidPlatformInjector.cs [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
+
+        public Stream GetBundledFileStream(string fileName)
         {
-            get
+            if (ExecutingAssembly != null)
+            {
+                var extendedName = $"{ExecutingAssembly.GetName().Name}.Resources.{fileName}";
+                return ExecutingAssembly.GetManifestResourceStream(extendedName);
+            }
+            else
```
Comment (wei-lee) [inline Core/Core.Platform.Android/AndroidPlatformInjector.cs]: If I am not mistaken, the `ExecutingAssembly` will always be set on app start, and it should never execute the `else` block?

File: Core/Core.Platform.Android/AndroidPlatformInjector.cs [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
-using AeroGear.Mobile.Core.Logging;
 
-[assembly: Xamarin.Forms.Dependency(typeof(AndroidPlatformInjector))]
-namespace AeroGear.Mobile.Core
+namespace AeroGear.Mobile.Core.Platform.Android
 {
-    /// <summary>
-    /// Class handles injection of Android specific classes and access to resoureces.
-    /// </summary>
     internal class AndroidPlatformInjector : IPlatformInjector
```
[the remark this answers is not in the document]
Comment (wojta) [reply Core/Core.Platform.Android/AndroidPlatformInjector.cs]: yes, I deleted that accidentally.

File: Core/Core.Platform.Android/packages.config [hunk 3]
```diff
@@ -1,21 +1,4 @@
...
-  <package id="Xamarin.Android.Support.Transition" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v4" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v7.AppCompat" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v7.CardView" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v7.MediaRouter" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v7.Palette" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.v7.RecyclerView" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Android.Support.Vector.Drawable" version="25.4.0.2" targetFramework="monoandroid71" />
-  <package id="Xamarin.Forms" version="2.5.0.280555" targetFramework="monoandroid71" />
+  <package id="NuGet.Build.Packaging" version="0.2.0" targetFramework="monoandroid71" developmentDependency="true" />
```
[the remark this answers is not in the document]
Comment (wojta) [reply Core/Core.Platform.Android/packages.config]: they aren't. At least not at the moment. We aren't using Xamarin.Forms and any UI.

File: Auth/Auth.Platform.Android/Auth.Platform.Android.csproj [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <OutputType>Library</OutputType>
     <RootNamespace>Auth.Platform</RootNamespace>
     <AssemblyName>Auth.Platform</AssemblyName>
-    <TargetFrameworkVersion>v8.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v7.1</TargetFrameworkVersion>
```
[the remark this answers is not in the document]
Comment (wojta) [reply Auth/Auth.Platform.Android/Auth.Platform.Android.csproj]: ok, thats a good idea

**Comment by secondsun** [comment]
Pulled it in and was able to get things to start up.  I say lets make the android installer its own PR an dmerge this guy.

**Pull request merged**